open-metadata-implementation/governance-servers/admin-services/admin-services-server/src/main/java/org/odpi/openmetadata/adminservices/OMAGServerVirtualizationService.java

```diff
@@ -56,14 +66,76 @@ public VoidResponse setVirtualizerConfig(String userId, String serverName, Virtu
+        final String connectionGUID           = UUID.randomUUID().toString();
+        final String endpointDescription      = "Virualization solution endpoint.";
+        final String connectionDescription    = "Virualization solution connection.";
+
+        String endpointName    = "Virtualizer.Endpoint." + serverName;
+
+        if ("org.odpi.openmetadata.openconnectors.governancedaemonconnectors.viewgenerator.virtulaizationsolutionconnector.derby".equals(virtualizationConfig.getVirtualizationProvider())){
+            try {
+                Map<String, Object> config = virtualizationConfig.getVirtualizationSolutionConfig();
+
+                endpoint.setType(Endpoint.getEndpointType());
+                endpoint.setGUID(endpointGUID);
+                endpoint.setQualifiedName(endpointName);
+                endpoint.setDisplayName(endpointName);
+                endpoint.setDescription(endpointDescription);
+                endpoint.setAddress(config.get("serverAddress").toString());
+
+                Map<String, Object> endpointProperties = new HashMap<>();
+                endpointProperties.put("username", config.get("username"));
+                endpointProperties.put("password", config.get("password"));
+                endpointProperties.put("timeoutInSecond", config.get("timeoutInSecond"));
+                endpointProperties.put("create", config.get("create"));
+
+                endpoint.setAdditionalProperties(endpointProperties);
+
+                String connectionName = "Vitualizer.Connection." + serverName;
+
+                Connection connection = new Connection();
+
+                connection.setType(Connection.getConnectionType());
+                connection.setGUID(connectionGUID);
+                connection.setQualifiedName(connectionName);
+                connection.setDisplayName(connectionName);
+                connection.setDescription(connectionDescription);
+                connection.setEndpoint(endpoint);
+
+                additionalProperties.put("logicTableName",config.get("logicTableName"));
+                additionalProperties.put("logicTableDefinition", config.get("logicTableDefinition"));
+                additionalProperties.put("getLogicTables", config.get("getLogicTables"));
+
+                connection.setAdditionalProperties(additionalProperties);
+
+                virtualizationConfig.setVirtualizationSolution(connection);
+
+            } catch (Exception e){
+                log.error("Error in loading virtualization solution configuration", e);
+            }
+        }
+
+        try {
             serverConfig.setVirtualizationConfig(virtualizationConfig);
 
             configStore.saveServerConfig(serverName, methodName, serverConfig);
-
-        } catch (OMAGInvalidParameterException e){
-            log.error("Invalid configuration! ", e);
+        } catch (Exception e){
+            log.error("Error save config.", e);
```

Comment:
Add proper AuditLog messages - debug logging is turned off in production.

Reply:
Hi Mandy, as you can see I've updated the PR with some auditlog. Could you please advise how I can pass the OMRSAuditLog object to the connector? For governance servers, we have the corresponding argument at initialize method, just like:

public void initialize(OpenLineageConfig openLineageConfig, OMRSAuditLog auditLog) 

But for connector, we don't have anything similar.

Thanks and have a nice day!

Reply:
The Connector needs a method where the audit log can be passed in.    The OMRS Connectors implement AuditableConnector which indicates to the OMRS to pass them an audit log object.  You need something similar for the virtualizer connectors